src/index.ts

```diff
@@ -134,6 +140,21 @@ export class Connection {
     }
   }
 
+  async transaction<T>(fn: (tx: Transaction) => Promise<T>): Promise<T> {
+    const tx = new Connection(this.config)
```

Comment:
What about transactions within transactions? I'm not sure how this translates to MySQL, but effectively snapshotting?

I'd expect this to reuse the existing connection, or to at minimum, copy over the session? I think we'd need to functionally test/confirm how this is expected to work and what is best, but pretty sure what we have here now is less than ideal. It'll work, but definitely would not supported nested transactions, it'll be an entirely new one even if you're already within one.

Reply:
We'll address nested transactions in a future pull request, but for now, we've gone ahead and refactored `Transaction` into it's own class that has it's own bespoke `execute` method and will reuse a connection.